lib/Doctrine/DBAL/Schema/Index.php

```diff
@@ -202,7 +202,7 @@ public function isFullfilledBy(Index $other)
     {
         // allow the other index to be equally large only. It being larger is an option
         // but it creates a problem with scenarios of the kind PRIMARY KEY(foo,bar) UNIQUE(foo)
-        if (count($other->getColumns()) != count($this->getColumns())) {
+        if (count($other->getColumns()) < count($this->getColumns())) {
```

Comment:
Not all databases are necessarily able to use larger indexes as a replacement for indexes with a subset of their covered columns. In addition to that, see the comment right above (about PK + UNIQUE)